src/System.Private.CoreLib/src/System/Runtime/InteropServices/ComActivator.cs

```diff
@@ -3,17 +3,20 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
+using System.Collections.Generic;
 using System.Diagnostics;
 using System.IO;
 using System.Reflection;
+using System.Runtime.Loader;
+using System.Threading;
 
 namespace System.Runtime.InteropServices
 {
     [ComImport]
     [ComVisible(false)]
     [Guid("00000001-0000-0000-C000-000000000046")]
     [InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
-    internal interface IClassFactory
+    public interface IClassFactory
```

Comment:
Why does this need to be public? This should be internal implementation detail.

Reply:
If the tests happen to need same definitions as what the internal implementation detail, we prefer to duplicate them in the tests.

Reply:
I get that and I would like to, but without enabling TypeEquivalence I am not going to be able to cast this definition of the interface to one I define in the test. This is the official COM definition of the interface but I don't really know how to fight the type system to make this work. Any suggestions? I can do the reflection approach, but since this is an official COM interface it seems like the better approach and is more straightforward.

Reply:
Hmm, I see.

It would useful to have comments on all `public` parts that are public just to make the testing of internals easier, but that are not called externally as part of the product.

Is `ComActivationContext` public just for testing purposes as well?

I am wondering whether all these public internal entrypoints should rather live in `Internal.Runtime.InteropServices`.

Reply:
>It would useful to have comments on all `public` parts that are public just to make the testing of internals easier, but that are not called externally as part of the product.

I can definitely do that. 

>Is `ComActivationContext` public just for testing purposes as well?

Yes.

> I am wondering whether all these public internal entrypoints should rather live in `Internal.Runtime.InteropServices` namespace.

I can do that if it will help indicate the intent more accurately.

Reply:
> I can do that if it will help indicate the intent more accurately.

It would match the public internal WinRT plumbing, e.g.: https://github.com/dotnet/coreclr/tree/master/src/System.Private.CoreLib/shared/Internal/Resources or https://github.com/dotnet/coreclr/tree/master/src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/WindowsRuntime